### PR TITLE
[dv/chip] Lc_walkthrough test part1

### PR DESCRIPTION
**Pull request by cindychip**
This PR adds one walkthrough test from RAW to ProdEnd state.
Upcoming PRs will walkthrough other paths of LC state transitions.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM

File: sw/device/tests/sim_dv/lc_walkthrough_test.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+static volatile const uint8_t kOtpUnlockToken[LC_TOKEN_SIZE] = {
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+    0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+};
+
+static void check_lc_state_transition_count(uint8_t exp_lc_count) {
+  LOG_INFO("Read LC count and check with expect_val=%0d", exp_lc_count);
+  uint8_t lc_count;
+  CHECK(dif_lc_ctrl_get_attempts(&lc, &lc_count) == kDifOk,
+        "Read lc_count register failed!");
```
Comment (timothytrippel) [inline sw/device/tests/sim_dv/lc_walkthrough_test.c:49]: I think you can just write:
```suggestion
  CHECK_DIF_OK(dif_lc_ctrl_get_attempts(&lc, &lc_count), "Read lc_count register failed!");
```

See [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/sca/lib/sca.h;drc=ca9ece08c9f7cd43a054980548a6bb94b86c903a;l=156).

File: sw/device/tests/sim_dv/lc_walkthrough_test.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+      LOG_INFO("Written and locked OTP secret0 partition!");
+      return true;
+    } else {
+      // Issue a LC state transfer to ProdEnd state.
+      dif_lc_ctrl_token_t token;
+      for (int i = 0; i < LC_TOKEN_SIZE; i++) {
+        token.data[i] = kLcExitToken[i];
+      }
+      CHECK(dif_lc_ctrl_mutex_try_acquire(&lc) == kDifOk);
```
Comment (timothytrippel) [inline sw/device/tests/sim_dv/lc_walkthrough_test.c:149]: same here:
```suggestion
      CHECK_DIF_OK(dif_lc_ctrl_mutex_try_acquire(&lc));
```

File: sw/device/tests/sim_dv/lc_walkthrough_test.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      dif_lc_ctrl_settings_t settings;
+      // TODO: randomize using external or internal clock.
+      settings.clock_select = kDifLcCtrlExternalClockEn;
+      CHECK(dif_lc_ctrl_transition(&lc, kDifLcCtrlStateProdEnd, &token,
```
Comment (timothytrippel) [inline sw/device/tests/sim_dv/lc_walkthrough_test.c:153]: same here, use `CHECK_DIF_OK`

File: sw/device/tests/sim_dv/lc_walkthrough_test.c [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+#define LC_TOKEN_SIZE 16
+
+static dif_lc_ctrl_t lc;
+static dif_otp_ctrl_t otp;
+
+/**
+ * Track LC state transition tokens.
+ *
+ * This tokens will be further randomized and override by DV sequence.
```
Comment (timothytrippel) [inline sw/device/tests/sim_dv/lc_walkthrough_test.c:24]: nit:
```suggestion
 * These tokens will be further randomized and overridden by the testbench.
```

**Review by timothytrippel — APPROVED** [review]

File: sw/device/tests/sim_dv/lc_walkthrough_test.c (hunk 4 above)
Comment (cindychip) [reply sw/device/tests/sim_dv/lc_walkthrough_test.c]: Thanks Tim for catching it, updated.

Comment (cindychip) [reply sw/device/tests/sim_dv/lc_walkthrough_test.c]: Thanks Tim for catching it, updated.

File: sw/device/tests/sim_dv/lc_walkthrough_test.c (hunk 1 above)
Comment (cindychip) [reply sw/device/tests/sim_dv/lc_walkthrough_test.c]: Thanks for the recommendation and reference! Updated

File: sw/device/tests/sim_dv/lc_walkthrough_test.c (hunk 2 above)
Comment (cindychip) [reply sw/device/tests/sim_dv/lc_walkthrough_test.c]: Thanks! Updated

Comment (cindychip) [reply sw/device/tests/sim_dv/lc_walkthrough_test.c]: Thanks! Updated

File: sw/device/tests/sim_dv/lc_walkthrough_test.c (hunk 3 above)
Comment (cindychip) [reply sw/device/tests/sim_dv/lc_walkthrough_test.c]: Thanks again! Updated

Comment (cindychip) [reply sw/device/tests/sim_dv/lc_walkthrough_test.c]: Thanks again! Updated

**Pull request merged**